lib/kaufmann_ex/test_support/mock_schema_registry.ex

```diff
@@ -29,7 +29,10 @@ defmodule KaufmannEx.TestSupport.MockSchemaRegistry do
   end
 
   @spec encode_decode(Request.t()) :: Event.t()
-  def encode_decode(%Request{format: format} = request) when format == :json or format == :avro do
+  def encode_decode(%Request{format: nil} = request),
+    do: encode_decode(%Request{request | format: :default})
+
+  def encode_decode(%Request{format: format} = request) do
     transcoder = Config.transcoder(format)
 
```

Comment:
Lets add an assertion that the format is defined. This will make error cases around unexpected formats easier to debug.

```elixir
    assert transcoder, """
      undefined format: #{format}
      Format must match a transcoder specified in Application env `:kaufmann_ex, :transcoder`
    """
```